dali/python/nvidia/dali/pipeline.py

```diff
@@ -1046,14 +1047,57 @@ def run(self):
         :meth:`share_outputs` and
         :meth:`release_outputs`
 
-        :return:
+        Parameters
+        ----------
+        pipeline_inputs :
+            Optional argument that can be used to provide inputs to DALI.
+            When DALI has any input operators defined (e.g. fn.external_source), you can provide the
+            inputs to those using named arguments in this function. The assumption is that
+            DALI pipeline has them defined and named properly::
+
+                @pipeline_def
+                def my_pipe():
+                    inp = fn.external_source(name="my_inp")
+                    return inp
+
+            With the example pipeline above, you can provide ``"my_inp"`` input into the
+            :meth:`run()` function::
+
+                p = my_pipe(prefetch_queue_depth=1, ...)
+                p.build()
+                p.run(my_inp=np.random((2,3,2)))
+
+            Such keyword argument specified in the :meth:`run()` function has to have a
+            corresponding input operator node declared in DALI pipeline.
+
+            As always when working with DALI, the value passed to the keyword argument has to
+            denote a whole batch of data.
+
+            Please note, that using this feature requires setting either ``prefetch_queue_depth=1``
+            or ``exec_pipelined=False`` in DALI Pipeline constructor.
+
+            This feature can be considered as a syntactic sugar over :meth:`feed_input` function.
+
+        Returns
+        -------
             A list of `TensorList` objects for respective pipeline outputs
         """
+        if len(pipeline_inputs) > 0 and (self.prefetch_queue_depth > 1 and self.exec_pipelined):
```

Comment:
Note: I wonder what's the value of `prefetch_queue_depth` in case of separated queues - and, specifically, what `prefetch_queue_depth > 1` would evaluate to. At least I'd try to specify separated queues and see the error below pops up.